### ISSUE TITLE
Serious intermittent bug on find & replace

### ISSUE THREAD

**Issue by samuelcolvin** [Microsoft/vscode#53338]
I didn't want to submit this until I would consistently reproduce it, however it's pretty serious and I can't reproduce it consistently although I've seen it 3 or 4 times today. (initially I thought I was going mad, but I'm not sure it's vscode).

Using the directory level find&replae I try to replace `organiz` with `organis`, with the following settings:

![image](https://user-images.githubusercontent.com/4039449/42098143-ec8e8952-7bb1-11e8-8586-6b9426f7b164.png)

vscode makes the following change (as seen from `git diff`):
```
-You can slowly build your confidence in maths through tutoring, and specify these skills in your resume when applying for a job. [Taking a look at online CV templates](https://novoresume.com/resume-templates) makes evident that these days, the very first thing you should include to attract prospective employers, are your skills. This is the perfect place to include any mathematical abilities you have which may be pertinent to the organization you wish to work for.
+organisce in maths through tutoring, and specify these skills in your resume when applying for a job. [Taking a look at online CV templates](https://novoresume.com/resume-templates) makes evident that these days, the very first thing you should include to attract prospective employers, are your skills. This is the perfect place to include any mathematical abilities you have which may be pertinent to the organization you wish to work for.
```

In other words, it substitutes `You can slowly build your confiden` for `organis` and leaves the actual `organiz` unchanged.

- VSCode Version:
```
Version 1.24.1
Commit 24f62626b222e9a8313213fb64b10d741a326288
Date 2018-06-13T17:47:35.732Z
Shell 1.7.12
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```
- OS Version: Ubuntu 18.04

Steps to Reproduce:
1. use the directory level find & replace for lots of files with the above settings
2. sometimes vscode messes up and deletes chunks of text you don't want to change.

Does this issue occur when all extensions are disabled?: Yes

**Comment by roblourens** [comment]
- Is there any chance that file is using CR-only line endings?
- What was happening with that file at the time, were you editing it? Was it open? Did it have unsaved changes? etc
- Does it happen in the same place a second time?

**Comment by samuelcolvin** [comment]
Tried to make it happen again, couldn't get it to happen in the same place. 

Very likely odd line endings. I think the file was last edited using netlify CMS which does many unholy things to line endings.

**Comment by samuelcolvin** [comment]
I'll try again and will provide the original file tomorrow.

**Comment by samuelcolvin** [comment]
Yes I've seen that. Will try to reproduce.

**Comment by samuelcolvin** [comment]
Ok, I've managed to narrow this down. It's not related to line endings but (I think to the length of the lines).

I've managed to replicate the problem with a single file in a directory, where the file contains only 1 line. Here's the problem occurring:

![image](https://user-images.githubusercontent.com/4039449/42134633-ad03fe20-7d37-11e8-88d6-8e83b0413447.png)

(If I go ahread and perform the find and replace the change is made wrongly as per the preview)

Here's the directory it occured with: [broken.zip](https://github.com/Microsoft/vscode/files/2152807/broken.zip)

Steps to reproduce:
* unzip the above, should be one directory `broken/` containing one file `testing.md`
* open vscode in the unziped directory with `code --disable-extensions broken/`
* click "search" on the right
* enter `inspiration` as the search term and `foobar` as the replacement term (I don't think either matter much as long as the search term is in the sentence)
* enable regular expression
* hit "enter" in the replacement term box to do the search
* the preview you see should be show the error

If you open the file in the editor or change the search vscode starts getting the replacement right and will then always get it right, you need to create a new file to get the problem to happen again.

**I don't want to be hyperbolic but this is a pretty serious bug (for a long time I couldn't believe it was actually happening) I'm not sure why no one else seems to have reported it, perhaps it's specific to Ubuntu or this exact version, but either way I think it needs to urgent attention.** 

(thankyou for vscode, it's generally great, sorry to be negative)

**Comment by roblourens** [comment]
That message in the preview, "Omitted long line with 1 match" is the key. We use ripgrep for search and it's behaving as if it has max-columns set, which vscode doesn't set.

Is it possible that you have a ripgrep config set up in your environment using the `RIPGREP_CONFIG_PATH` environment var? If so, this would be fixed in the next release.

**Comment by samuelcolvin** [comment]
BINGO!

`.ripgreprc`:
```
--max-columns=150
--smart-case
```

**Comment by roblourens** [comment]
Could you try the insiders build to verify that this is already fixed? (It should now be correctly ignoring that file) https://code.visualstudio.com/insiders/

**Comment by samuelcolvin** [comment]
seems to be fixed by insider. Thank you.

**Comment by roblourens** [comment]
Thanks for helping to troubleshoot it, sorry for the issue.
